### PR TITLE
add recommendations on reuse of ECDH keys

### LINKED ISSUE #267: Strengthen Recommendation about ECDH Exponent Reuse
Ilari Liusvaara claims regarding Section 6.4 that reusing ECDH exponents is insecure unless one either:

1) Checks for point validity, or
2) Uses montgomery ladder with twist-secure curve.

He further states:

"Currently the only curves in TLS satisfying 2) are x25519 and x448. For
any other elliptic curve, reusing exponents without validating points
is insecure. Few thousand connection attempts at most is sufficient
for recovering the ECDH private key, which compromises any connection
that used it."

### PR DESCRIPTION
**Pull request by thomas-fossati**
Fixes #267

### CODE REVIEW

File: BCP195bis/draft-ietf-uta-rfc7525bis.md [hunk 1]
```diff
@@ -776,13 +814,16 @@ For performance reasons, many TLS implementations reuse Diffie-Hellman and Ellip
...
+* An "invalid curve" attack can be mounted against elliptic-curve DH if the victim does not verify that the received point lies on the correct curve.  If the victim is reusing the DH secrets, the attacker can repeat the probe varying the points to recover the full secret (see {{Antipa2003}} and {{Jager2015}}).
 
-To address these concerns, TLS implementations SHOULD NOT use static DH keys and SHOULD NOT reuse ephemeral DH keys across multiple connections.
+To address these concerns:
 
-<cref>TODO: revisit when draft-bartle-tls-deprecate-ffdhe becomes a TLS WG item, since it specifies MUST NOT rather than SHOULD NOT.</cref>
+* TLS implementations SHOULD NOT use static finite-field DH keys and SHOULD NOT reuse ephemeral finite-field DH keys across multiple connections.
+* Implementations that want to reuse elliptic-curve DH keys SHOULD either use a "safe curve" {{SAFECURVES}} (e.g., X25519), or perform the checks described in {{NIST.SP.800-56A}} on the received points.
```
[the remark this answers is not in the document]
Comment (thomas-fossati) [reply BCP195bis/draft-ietf-uta-rfc7525bis.md:826]: ack

[the remark this answers is not in the document]
Comment (thomas-fossati) [reply BCP195bis/draft-ietf-uta-rfc7525bis.md:826]: s/Implementations/Server implementation/

File: BCP195bis/draft-ietf-uta-rfc7525bis.md [hunk 2]
```diff
@@ -776,13 +814,16 @@ For performance reasons, many TLS implementations reuse Diffie-Hellman and Ellip
...
+* TLS implementations that reuse exponents should test the DH public key they receive for group membership, in order to avoid some known attacks. These tests are not standardized in TLS at the time of writing, although general guidance in this area is provided by {{NIST.SP.800-56A}} and implemented in many popular stacks.
+
+* Under certain conditions, the use of static finite-field DH keys, or of ephemeral finite-field DH keys that are reused across multiple connections, can lead to timing attacks (such as those described in {{RACCOON}}) on the shared secrets used in Diffie-Hellman key exchange.
 
-* Under certain conditions, the use of static DH keys, or of ephemeral DH keys that are reused across multiple connections, can lead to timing attacks (such as those described in {{RACCOON}}) on the shared secrets used in Diffie-Hellman key exchange.
+* An "invalid curve" attack can be mounted against elliptic-curve DH if the victim does not verify that the received point lies on the correct curve.  If the victim is reusing the DH secrets, the attacker can repeat the probe varying the points to recover the full secret (see {{Antipa2003}} and {{Jager2015}}).
```
[the remark this answers is not in the document]
Comment (thomas-fossati) [reply BCP195bis/draft-ietf-uta-rfc7525bis.md:821]: yes

File: BCP195bis/draft-ietf-uta-rfc7525bis.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (thomas-fossati) [reply BCP195bis/draft-ietf-uta-rfc7525bis.md]: Done in 37ad8be

**Review by stpeter — APPROVED** [review]

**Pull request merged**